yt/yt/server/queue_agent/queue_static_table_exporter.cpp

```diff
@@ -773,42 +746,87 @@ TQueueExportProfilingCounters::TQueueExportProfilingCounters(const TProfiler& pr
     IInvokerPtr invoker,
     IAlertCollectorPtr alertCollector,
     const TProfiler& queueProfiler,
     const TLogger& logger)
     : ExportName_(std::move(exportName))
+    , Queue_(std::move(queue))
+    , Config_(config)
+    , ExportProgress_(New<TQueueExportProgress>())
     , ClientDirectory_(std::move(clientDirectory))
     , Invoker_(std::move(invoker))
     , AlertCollector_(std::move(alertCollector))
     , ProfilingCounters_(New<TQueueExportProfilingCounters>(queueProfiler.WithPrefix("/static_export").WithTag("export_name", ExportName_)))
     , Logger(logger.WithTag("ExportName: %v", ExportName_))
 { }
 
-TFuture<void> TQueueExporter::RunExportIteration(
-    const TCrossClusterReference& queue,
-    const TQueueStaticExportConfig& exportConfig)
+TFuture<void> TQueueExporter::RunExportIteration()
 {
+    auto config = GetConfig();
+
     auto exportTask = New<TQueueExportTask>(
-        ClientDirectory_->GetClientOrThrow(queue.Cluster),
+        ClientDirectory_->GetClientOrThrow(Queue_.Cluster),
         Invoker_,
         ProfilingCounters_,
-        queue.Path,
-        exportConfig,
+        Queue_.Path,
+        config,
         Logger);
+
     auto exportTaskFuture = exportTask->Run();
-    exportTaskFuture.Subscribe(BIND([this, this_ = MakeStrong(this)] (const TError& error) {
-        if (!error.IsOK()) {
+
+    exportTaskFuture.Subscribe(BIND([
+        this,
+        this_ = MakeStrong(this),
+        config = std::move(config)
+    ] (const TErrorOr<TQueueExportProgressPtr>& exportProgress) {
+        if (!exportProgress.IsOK()) {
             AlertCollector_->StageAlert(CreateAlert(
                 NAlerts::EErrorCode::QueueAgentQueueControllerStaticExportFailed,
                 "Failed to perform static export for queue",
                 /*tags*/ {{"export_name", ExportName_}},
-                error));
+                /*error*/ exportProgress));
+        } else {
+            // NB(apachee): This assumes that queue exporter owns the returned pointer,
+            // and it shouldn't be modified by anyone else.
+            auto nextExportProgress = exportProgress.Value();
+
+            auto guard = Guard(Lock_);
+
+            if (config == Config_ && nextExportProgress->LastExportedFragmentUnixTs > ExportProgress_->LastExportedFragmentUnixTs) {
+                ExportProgress_ = std::move(nextExportProgress);
+            }
         }
         AlertCollector_->PublishAlerts();
     }));
-    return exportTaskFuture;
+
+    return exportTaskFuture.AsVoid();
+}
+
+void TQueueExporter::OnConfigUpdated(const NQueueClient::TQueueStaticExportConfig& config) {
+    auto guard = Guard(Lock_);
+
+    if (config.ExportDirectory != Config_.ExportDirectory) {
+        ExportProgress_ = New<TQueueExportProgress>();
+    }
+
+    Config_ = config;
+}
+
+TQueueExportProgressPtr TQueueExporter::GetExportProgress() const
+{
+    auto guard = Guard(Lock_);
+    auto exportProgress = CloneYsonStruct(ExportProgress_);
```

Comment:
I thought it is better to prevent any possible issues, when the caller changes the value, at the expense of little performance degradation. In my opinion, making the caller responsible for that overcomplicates things, because that would mean more potential points of failure, even though, probably, this method wouldn't be called in many places.